SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -349,33 +363,41 @@ Interest InterestFactory::CreatePlayerOwnedActorInterest() const
 
 			if (SpatialGDKSettings->bEnableClientResultTypes)
 			{
-				NewQuery.ResultComponentId = ClientResultType;
+				NewQuery.ResultComponentId = ClientNonAuthInterestResultType;
 			}
 			else
 			{
 				NewQuery.FullSnapshotResult = true;
 			}
 
-			ClientComponentInterest.Queries.Add(NewQuery);
+			AddComponentQueryPairToInterestComponent(OutInterest, ClientEndpointComponentId, NewQuery);
 		}
 	}
+}
 
-	Interest NewInterest;
-	// Server Interest
-	if (SystemConstraints.IsValid() && SpatialGDKSettings->bEnableServerQBI)
-	{
-		NewInterest.ComponentInterestMap.Add(SpatialConstants::POSITION_COMPONENT_ID, ServerComponentInterest);
-	}
-	// Client Interest
-	if (ClientConstraint.IsValid())
+void InterestFactory::AddClientSelfInterest(Interest& OutInterest) const
+{
+	Query NewQuery;
+	// Just an entity ID constraint is fine, as clients should not become authoritative over entities outside their loaded levels
+	NewQuery.Constraint.EntityIdConstraint = EntityId;
+
+	NewQuery.ResultComponentId = ClientAuthInterestResultType;
+
+	AddComponentQueryPairToInterestComponent(OutInterest, SpatialConstants::GetClientAuthorityComponent(GetDefault<USpatialGDKSettings>()->bUseRPCRingBuffers), NewQuery);
+}
+
+void InterestFactory::AddComponentQueryPairToInterestComponent(Interest& OutInterest, const Worker_ComponentId ComponentId, const Query& QueryToAdd)
+{
+	if (!OutInterest.ComponentInterestMap.Contains(ComponentId))
 	{
-		NewInterest.ComponentInterestMap.Add(SpatialConstants::GetClientAuthorityComponent(SpatialGDKSettings->bUseRPCRingBuffers), ClientComponentInterest);
+		ComponentInterest NewComponentInterest;
+		OutInterest.ComponentInterestMap.Add(ComponentId, NewComponentInterest);
 	}
-
-	return NewInterest;
+	auto& Queries = OutInterest.ComponentInterestMap[ComponentId].Queries;
```

Comment:
Can we just modify the `Queries` directly?

Reply:
sure